mlrun/projects/project.py

```diff
@@ -789,6 +802,26 @@ def __init__(
         self._status = None
         self.status = None
 
+        if any(
+            [
+                name,
+                description,
+                params,
+                functions,
+                workflows,
+                artifacts,
+                artifact_path,
+                conda,
+                default_requirements,
+            ]
+        ):
+            # TODO: remove in 1.5.0 along with all arguments except metadata and spec
+            warnings.warn(
+                "Project constructor arguments are deprecated in 1.3.1 and will be removed in 1.5.0,"
```

Comment:
hmm, i dont think it it is legit to drepcate things in 1.3.1 and remove on 1.5.x, because customers with 1.3.0 wont see that warning, but only @ 1.4.0 and then will have one version to handle, which is oppose to our statement of two versions.
I suggest to
1. move pr to be against 1.4.0
2. deprecate on 1.6.0